### DISCUSSION TITLE
Feature request: Support for separate, non-STATE related mqtt topics

### DISCUSSION THREAD

**Discussion by AleXSR700** [arendst/Tasmota#19478] [Ideas]
Hello everyone,
I know from discord discussions that this will not be an easy sell. But I will try.

So, tasmota is the main source of sensor data for a lot of people.
But the stock capabilites, especially for the primary ESP8266 chip, are limited when it comes to data processing. This is totally understandable due to hardware restrictions. Nevertheless, complex rules are unstable and will not behave as created (discussed on discord, so I will not spam here unless asked to.

The alternative to tasmota's built-in approaches is external processing of the data on more powerful systems like an RPi e.g. running Node-RED. Possibilites are more or less endless there.

However, tasmota publishes all sensor information on the telemetry topic. Makes sense, yes. But if you want to process the data before it is received by your home automation server, you must prevent that home automation server from subscribing to the telemetry topic of tasmota.
That is generally not possible without changing the source code of those servers (sometimes they are open source, sometimes not).

The easy way of handling this is to changing the topic tasmota uses, e.g. by changing `Prefix3`.

BUT, and this is the pitfall right now, tasmota then of course also sends `LWT` and `STATE` on the `Prefix3` topic, which kills all functionality of TDM or any home automation server. Meaning, the devices are no longer discovered and you lose all functionality (not just the sensor data).

The solution would be to add a dedicated setting, e.g. `Prefix4` which only applies to telemetry data != `STATE` || `LWT`. Those are the only two that are required for basic functionality of any remote subscriber.

So be it BLE data or SENSOR or whatever else there might be, could be changed to `Prefix4` whilst /tele/STATE and tele/LWT remain on `Prefix3`.

I hope it is not a huge code change because it would be an elegant enough solution to a world of post-processing whilst retaining the basic control functions :)

Please consider my request and let's open up the discussion :)

Thank you for your continued support
Alex

**Comment by barbudor** [comment]
If you change Prefix3 to allow SENSOR to be managed by NodeRed or other, then you can also have NodeRed to reforward LWT and STATE, just a couple of nodes

**Reply by barbudor** [reply]
What I don't get is that if you want to do that processing on your server, why aren't you doing it in your home automation system ?
It should be configurable to handle that, and they don't have the excuse to run on limited hardware

**Reply by AleXSR700** [reply]
I am already relaying the LWT and STATE, as well as the 3 INFO topics.
But since those are not sensor telemetry data of the actual measurements, they would not need to be part of the same prefix as the sensor telemetry data.

Home Assistant does not allow the treatment of incoming messages unless you disable e.g. the built-in tasmota, create one template sensor per entity and then filter there.
BUT that means creating hundreds of template sensors. Which would be the worst possible approach.

Just maybe as an additional info:
I already have everything set up in Node-RED to handle this issue. But since this is an open source project I thought I would suggest this improvement to also aid others in making the switch from Rules etc. to external processing.

It is kind of an improvement suggestion :)

**Reply by barbudor** [reply]
> BUT that means creating hundreds of template sensors

But you are creating them in NR anyhow don't you?

What kind of processing would you do on the SENSOR message that wouldn't break it's structure as announced by the discovery protocol?

**Reply by sfromis** [reply]
The STATE payloads are very much telemetry data about the Tasmota device. Telemetry does not merely mean sensors. Separating this from sensor data is just your thing for a particular use case, instead of something inherently different.
